charts/opensearch-dashboards/values.yaml

```diff
@@ -146,11 +146,12 @@ resources:
     memory: "512M"
 
 autoscaling:
-  enabled: false
+  # This requires metrics server to be installed, to install use kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml
+  # See https://github.com/kubernetes-sigs/metrics-server
+  enabled: true
   minReplicas: 1
-  maxReplicas: 100
+  maxReplicas: 10
   targetCPUUtilizationPercentage: 80
-  # targetMemoryUtilizationPercentage: 80
```

Comment:
Can we add memory as well apart from CPU here because this is something I have seen clusters experiencing. Though I am not really sure.

@peterzhuamazon @DandyDeveloper What's your opinion on this?

Reply:
Hey @TheAlgo, @DandyDeveloper, @peterzhuamazon 
I have added memory as well, I have seen cluster issues with pods getting terminated OOM killed error, so I have pushed hpa with api Autoscaling/v1, this is stable version that has only CPU based scaling, the other supported API is Autoscaling/v2beta2 this has both CPU and memory but in beta version, but from doc I see its now in stable release https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/ (v2beta1 is deprecated), so we should be good to use this v2beta2 API, I have added targetMemoryUtilizationPercentage in my latest commit.
<img width="992" alt="Screen Shot 2022-03-20 at 7 09 32 AM" src="https://user-images.githubusercontent.com/28733332/159166844-38d35330-ffb2-417e-a510-52a132e5a16d.png">

Reply:
@prudhvigodithi @TheAlgo For opensearch-dashboards I don't see this as an issue. 

What's the feature state and planned implementation for `v1/v2beta2`? I can't find any reference on when it was introduced.  https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#horizontalpodautoscaler-v2beta2-autoscaling

We might want to be careful as some people using old clusters won't be able to use the HPA if it's only available in 1.22+ (But I can't find exactly when this was introduced).

Can we triple check that? Otherwise I'd keep CPU targets for now and use `v1`

Reply:
Hey @DandyDeveloper, good point, I can see this v2beta2 was introduced in k8s 1.12 release notes
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.12.md

The cluster which I have tested is v1.21.5 that has all autoscaling/v1, autoscaling/v2beta1, autoscaling/v2beta2
```
kubectl api-versions | grep autoscaling
autoscaling/v1
autoscaling/v2beta1
autoscaling/v2beta2
```

v2beta1 is deprecated in 1.19 and you should use v2beta2 going forward. https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.19.md

Also on Default k8s docs I can see this 
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/
FEATURE STATE: Kubernetes v1.23 [stable]
(the autoscaling/v2beta2 API version previously provided this ability as a beta feature)

But looks like GKE has some issues and resolved in 1.15.5
https://issuetracker.google.com/issues/135624588?pli=1

So to summarize autoscaling/v2beta2 should be available from k8s version 1.12, GKE 1.15.5.
I'm ok to go with v1 and keep CPU alone to scale excluding memory for now, assuming there are chances some users can be before 1.12 version, or I can go with autoscaling/v2beta2 adding a note saying that it should be above 1.12 for this to work @TheAlgo and @DandyDeveloper

Reply:
@prudhvigodithi In that case lets keep CPU for now. Lets not add that, it might create unnecessary confusion for the users.

Reply:
@prudhvigodithi ^^

Reply:
Hey @peterzhuamazon and @TheAlgo I have reverted to v1, let's merge this and open an issue to add even memory as a scaler metric, and we can add updates to that issue about v2 autoscaling version. 
Thank you

Reply:
Hey @alborotogarcia and @DandyDeveloper, can you please check this change that I have pushed regarding v1.
Thank you